### ISSUE TITLE
Make data errors comprehensive

### ISSUE THREAD

**Issue by hmgaudecker** [janosg/skillmodels#42]
My excuses if this is already fixed in the current release, I am still on the one from late August (aside: Would be great to see tags in the repo and to be able to do `skillmodels.__version__`, then I would be more precise here, too :-))

Upon creation, `SkillModel()` is checking for variation in the measurements per period, which is great. However, it would be even better if it did so for all of them rather than one-by-one. With a large dataset, it is really slow to go through all of this. I did a bit of manual work to speed it up, but still 

In an ideal world, the error message would look like the params Dataframe with the subset of variable/period combinations that are violating the constraints, i.e., from the top of my head,

```
factor      measurement    period    problem

cognition   abc            1         All values equal to 1.0
            cde            5         All values equal to 0.0
            xyz            2         All values missing 

non-cong    abc            1         All values equal to 1.0
```
This would allow one to go quickly through the model specification to eliminate all problems in one go.

**Comment by janosg** [comment]
This is a very good idea!

In the current version we only check for variables that are all NaN because this is the most frequent case. Also, this will now always raise an error, whereas in previous versions the user could specify to automatically drop that variable. I think this small convenience was not worth the complexity it introduced. 

The code that generates the report should be in `data_processing`, not like currently in `model_processing`.

For the versioning I opened #43

**Comment by hmgaudecker** [comment]
So the check for any variation has disappeared between August and now? That would be important to know because it happens frequently in my application if I follow kitchen sink-type approaches to select variables...

**Comment by hmgaudecker** [comment]
Welcome to the world of ICD codes 😁

**Comment by janosg** [comment]
Then this should probably be the next task for @mpetrosian

**Comment by hmgaudecker** [comment]
Even before that, same idea: Please check as the very first thing whether all data columns that will be needed are present in the data. I am just going KeyError by KeyError through typos...

**Comment by janosg** [comment]
And use something like [this](https://github.com/OpenSourceEconomics/estimagic/blob/cdd25b1c375cbfaca46f23176957f69041a203e5/estimagic/optimization/utilities.py#L116) to propose a few similarly named variable in the dataset. Fuzzywuzzy is already available in skillmodels as a dependency of estimagic.

**Comment by janosg** [comment]
Solved in #47
